libyul/optimiser/StackToMemoryMover.cpp

```diff
@@ -0,0 +1,210 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include <libyul/optimiser/StackToMemoryMover.h>
+#include <libyul/optimiser/NameDispenser.h>
+#include <libyul/backends/evm/EVMDialect.h>
+
+#include <libyul/AsmData.h>
+
+#include <libsolutil/CommonData.h>
+#include <libsolutil/Visitor.h>
+
+using namespace std;
+using namespace solidity;
+using namespace solidity::yul;
+
+namespace
+{
+void appendMemoryStore(
+	vector<Statement>& _statements,
+	langutil::SourceLocation const& _loc,
+	YulString _mpos,
+	Expression _value
+)
+{
+	_statements.emplace_back(ExpressionStatement{_loc, FunctionCall{
+		_loc,
+		Identifier{_loc, "mstore"_yulstring},
```

Comment:
Maybe define the name of the mstore function in the dialect?

Reply:
The step is currently restricted to "EVM with objects" anyways - but I can still do that, yes.